File: typescript/pe002.ts
function pe002(max02:number): number {
  let total:number = 0
  let previous:number = 1
  let current:number = 1

  while (current < max02)
  {
    let next = current + previous
    if (!(current % 2))
      total += current
    previous = current
    current = next
  }

  return total
}

const max02:number = 4000000
console.log(pe002(max02))

File: c++/pe002.cpp
// This is a solution for http://projecteuler.net/problem=2

#include <cstdio>

int main() {
  const int max = 4000000;

  int total = 0;
  int previous = 1;
  int current = 1;
  while (current < max) {
    int next = previous + current;
    if (!(current % 2))
      total += current;
    previous = current;
    current = next;
  }
  printf("%d\n", total);
  return(0);
}
